src/lib/Types/Primitives.hs

```diff
@@ -140,6 +140,10 @@ data PrimOp e =
       -- Pointer to the stdout-like output stream
       | OutputStreamPtr
       | SynthesizeDict SrcPosCtx e  -- Only used during type inference
+      | ProjMethod e e  -- project a method, by source name, from the dict
```

Comment:
There's no source name here, so the comment is a bit confusing. Perhaps clarify that you want a `Label` in the first argument?